### PR TITLE
feat: [SCM-36]: Refactor SCM to Code

### PR DESCRIPTION
**Pull request by tan-nhu**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by tan-nhu** [comment]
retrigger titlecheck

**Review by mtran7 — APPROVED** [review]

File: src/index.html [hunk 1]
```diff
@@ -33,11 +33,13 @@
...
+        let url = window.apiUrl ? `${window.apiUrl}/${str}` : window.location.pathname.replace('ng/', '') + str
+
         if (window.browserRouterEnabled) {
-          return `${window.apiUrl || ''}/${str}`
-        } else {
-          return window.apiUrl ? `${window.apiUrl}/${str}` : window.location.pathname.replace('ng/', '') + str
+          url = `${window.apiUrl || ''}/${str}`
         }
+
+        return url.includes('remoteEntry.js') ? `${url}?timestamp=${Date.now()}` : url
```
Comment (johannesHarness) [inline src/index.html:42]: could it ever be the case that `url` already has parameters, so adding `?timestamp=...` will lead to two `?`?

**Review by cjlee01 — APPROVED** [review]

File: src/index.html (hunk 1 above)
Comment (tan-nhu) [reply src/index.html:42]: No, remoteEntry.js never has a param.

File: src/modules/10-common/navigation/ModuleList/NavModule/NavModule.tsx [hunk 2]
```diff
@@ -35,7 +35,7 @@ const navModuleToClassMap: Record<NavModuleName, string> = {
   [ModuleName.CE]: css.ccm,
   [ModuleName.STO]: css.sto,
   [ModuleName.CHAOS]: css.chaos,
-  [ModuleName.SCM]: css.default
+  [ModuleName.CODE]: css.default
```
Comment (johannesHarness) [inline src/modules/10-common/navigation/ModuleList/NavModule/NavModule.tsx:38]: I assume we'll have to add `css.code` at some point?

Comment (tan-nhu) [reply src/modules/10-common/navigation/ModuleList/NavModule/NavModule.tsx:38]: We don't have custom CSS for code nav. Hence the default.

File: src/modules/60-code/components/SideNav/SideNav.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+  const projectSelectHandler: ProjectSelectorProps['onSelect'] = data => {
+    updateAppStore({ selectedProject: data })
+
+    history.push(
+      routes.toCODERepositoriesListing({ space: [accountId, data.orgIdentifier as string, data.identifier].join('/') })
+    )
+  }
+  const isCommits = useMemo(() => routeMatch.path.includes(codePathProps.commitRef), [routeMatch])
+  const isBranches = useMemo(() => routeMatch.path.includes(codePathProps.branch), [routeMatch])
+  const isSettings = useMemo(() => routeMatch.path.endsWith('/:repoName/settings'), [routeMatch])
```
Comment (johannesHarness) [inline src/modules/60-code/components/SideNav/SideNav.tsx:36]: should this also become a constant in `codePathProps`?

**Review by johannesHarness — APPROVED** [review]

File: src/modules/60-code/components/SideNav/SideNav.tsx (hunk 3 above)
Comment (tan-nhu) [reply src/modules/60-code/components/SideNav/SideNav.tsx:36]: We'll refine this route later on. Right now it's a place-holder for Sahithi to work on her webhook prototype.

**Comment by tan-nhu** [comment]
FM to make the changes available for same refactoring in Gitness - https://github.com/harness/gitness/pull/92. Will work on the coverage later on.

**Pull request merged**